### DISCUSSION TITLE
使用动态主题，如何配合增加定制主题（因为ie不支持var）

### DISCUSSION THREAD

**Discussion by Atw-Lee** [ant-design/ant-design#46161] [Q&A] [answered]
antd^ 4

需求：想要使用动态主题，但因为ie不支持var，还需要定制主题
引入了import "antd/dist/antd.variable.min.css";

打包之后的产物：
<img width="368" alt="image" src="https://github.com/ant-design/ant-design/assets/95407716/536b9e6f-8ad2-40df-875e-0d929f43232f">
我想要定制主题用来修改1890ff，该如何做呢。

我尝试引用import 'antd/dist/antd.variable.less';配合less-loader，但是打包之后动态主题的样式就没有了。
我还尝试使用postcss，也没有解决。

**Comment by MadCcc** [comment, marked as answer]
v4 不使用 css 变量的话无法实现动态主题，只能修改 less 变量进行预构建

**Reply by Atw-Lee** [reply]
修改 less 变量进行预构建，产物当中没有var了，是不是就无法动态主题了

**Reply by MadCcc** [reply]
是的，只能动态替换 css 文件
